### ISSUE TITLE
False positive in `unnecessary_nesting_linter()` for `test_that()` calls

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2387]
``` r
library(lintr)

'test_that(
  desc = "addition works",
  code = {
    expect_identical(1L + 1L, 2L)
  }
)' -> code

lint(text = code, linters = unnecessary_nesting_linter())
#> <text>:3:10: warning: [unnecessary_nesting_linter] Reduce the nesting of this statement by removing the braces {}.
#>   code = {
#>          ^
```

<sup>Created on 2023-12-04 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

---

Note that removing the braces will generate a `{testthat}` warning in interactive session:

```r
library(testthat)
options(warn = 2L)
test_that(
  desc = "addition works",
  code = expect_identical(1L + 1L, 2L)
)
#> Error: (converted from warning) The `code` argument to `test_that()` must be a braced expression to get accurate file-line information for failures.
```

**Comment by IndrajeetPatil** [comment]
Actually, it looks like what the throws the linter off is not the keyword arguments but rather braces being on their own lines:

``` r
library(lintr)

'test_that("addition works",
  {
    expect_identical(1L + 1L, 2L)
  }
)' -> code

lint(text = code, linters = unnecessary_nesting_linter())
#> <text>:2:3: warning: [unnecessary_nesting_linter] Reduce the nesting of this statement by removing the braces {}.
#>   {
#>   ^

'test_that("addition works", {
  expect_identical(1L + 1L, 2L)
})' -> code

lint(text = code, linters = unnecessary_nesting_linter())
```

<sup>Created on 2023-12-04 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
Closing as duplicate of #2334
